src/super_gradients/common/factories/base_factory.py

```diff
@@ -0,0 +1,41 @@
+from typing import Union, Mapping
+
+
+class AbstractFactory:
+    """
+    An abstract factory to generate an object from a string or a dictionary
+    """
+
+    def get(self, conf: Union[str, dict]):
+        """
+        Get an instantiated object.
+            :param conf: a configuration. either a type name (str) or a dictionary {type_name(str): {parameters...}}
+        """
+        raise NotImplementedError
+
+
+class BaseFactory(AbstractFactory):
+
```

Comment:
This is a bit hard to follow, can you please add documentation that's a bit more detailed ?
Also im not sure I understand properly what the behaviour would be for instantiated objects inside a list ( such as train_metrics_list = [Accuracy(), Top5()]).

It is important we don't limit our users to only use our built in objects...

Reply:
added documentation.
list factory allows the user to generate lists as well (explained in docs)